kernel/memory/src/paging/mapper.rs

```diff
@@ -82,14 +89,13 @@ impl Mapper {
     /// and also shows their `PteFlags`.
     /// 
     /// The page table details are written to the the given `writer`.
-    pub fn dump_pte<W: Write>(&self, writer: &mut W, virtual_address: VirtualAddress) -> fmt::Result {
+    pub fn dump_pte(&self,virtual_address: VirtualAddress) {
```

Comment:
is there a hard reason for this change? It directly printed to the log ages ago, but I thought adding the `Write`r param made it more flexible.

Reply:
AFAICT previously you needed to add a dependency on the `logger` crate and call it like this:
```rust
dump_pte(&mut logger::LOGGER.lock().unwrap().writers[0], virtual_address);
```
and I don't know where else you could write the output feasibly.

Reply:
well, the main use case would be writing it to a String, at which point you could do anything with it.

That does raise the related question of how we can get access to the current logger as a stream, which is apparently pretty shit rn.

Reply:
I can change it back, considering it's irrelevant to the PR, but it does seem strange that it's the only place in Theseus that uses a `Write` trait bound rather than `log`. E.g. `dump_frame_allocator_state` just prints to `log`.

Reply:
yea, i agree it is strange, especially considering it's a one-off.
It's fine, leave the change in ig, since this is a larger issue about being able to treat the logger stream as a `fmt::Write`r.